src/data/actions/assets.js

```diff
@@ -20,9 +20,14 @@ export const requestAssetsFailure = response => ({
   data: response,
 });
 
+export const requestingAssets = () => ({
+  type: assetActions.request.REQUESTING_ASSETS,
+});
+
 export const getAssets = (request, courseDetails) =>
-  (dispatch, getState) =>
-    clientApi.requestAssets(courseDetails.id, {
+  (dispatch, getState) => {
+    dispatch(requestingAssets());
```

Comment:
I think this makes sense :+1: we can later check for this state and display some sort of loading spinner too.